xlators/cluster/dht/src/dht-selfheal.c

```diff
@@ -1792,7 +1792,7 @@ dht_selfheal_layout_new_directory(call_frame_t *frame, loc_t *loc,
                      chunk);
     } else {
         weight_by_size = _gf_false;
-        chunk = ((unsigned long)0xffffffff) / bricks_to_use;
+        chunk = ((unsigned long)0xffffffff) / ((double)bricks_to_use);
```

Comment:
```suggestion
        chunk = ((double)0xffffffff) / ((double)bricks_to_use);
```

Reply:
Thanks @xhernandez. Will make the changes. :)